addons/account/models/company.py

```diff
@@ -155,6 +156,17 @@ def _check_fiscalyear_last_day(self):
             if rec.fiscalyear_last_day > max_day:
                 raise ValidationError(_("Invalid fiscal year last day"))
 
+    @api.depends('vat', 'country_id')
+    def _compute_company_registry(self):
+    # If a belgian company has a VAT number then it's company registry is it's VAT Number (without country code).
+        for company in self:
+            if company.country_id.code == 'BE' and company.vat:
+                ResPartner = self.env['res.partner']
+                vat_country, vat_number = ResPartner._split_vat(company.vat)
+                # set value only if VAT number is valid
+                if vat_country == 'be' and ResPartner.simple_vat_check(vat_country, vat_number):
+                    company.company_registry = vat_number
```

Comment:
wtf? This is a country specific code 
```
    @api.depends('vat')
    def _compute_company_registry(self):
        for company in self:
            if company.vat:
                ResPartner = self.env['res.partner']
                vat_country, vat_number = ResPartner._split_vat(company.vat)
                # set value only if VAT number is valid
                if ResPartner.simple_vat_check(vat_country, vat_number):
                    company.company_registry = vat_number
                else:
                    company.company_registry = company.company_registry
            else:
                company.company_registry = company.company_registry
```
Note: A compute must always assign explicitely a value